packages/preact/src/hooks/useSwatches/sortOptions.ts

```diff
@@ -0,0 +1,54 @@
+import { SwatchOption } from "../types"
+
+const SIZE_ORDER = [
+  "4XS",
+  "3XS",
+  "2XS",
+  "XXS",
+  "XS",
+  "S",
+  "M",
+  "L",
+  "XL",
+  "XXL",
+  "2XL",
+  "XXXL",
+  "3XL",
+  "4XL",
+  "One Size"
+]
+
+function getSortIndex(value: string, field: string) {
+  const num = parseFloat(value)
+  if (!isNaN(num)) return num
+
+  if (field === "size") {
+    const index = SIZE_ORDER.indexOf(value)
+    return index === -1 ? SIZE_ORDER.length : index
+  }
+
+  return Number.MAX_SAFE_INTEGER
+}
+
+export function sortOptions(field: string, options: SwatchOption[]) {
+  if (field === "size") {
+    return [...options].sort((a, b) => {
```

Comment:
can't you do something like this instead:

sort((a, b) => {
  const [aa, bb] = [a, b].map(normalize)
  return aa < bb ? -1 : 1
})